dist/stat_data.py

```diff
@@ -916,7 +916,6 @@ def __init__(self, name, desc, flags=''):
     ##########################################
     # Tiered storage statistics
     ##########################################
-    StorageStat('tiered_object_size', 'tiered storage object size', 'no_clear,no_scale,size'),
```

Comment:
@sueloverso When backporting this change, I found that these statistics live in connection and data source statistics in mongodb-6.0. On develop these statistics are in connection stats section. Should we be moving the other statistics to connection stats as well?

Reply:
Those statistics were removed in WT-8338. If the stats are a problem then that should be backported first. Backporting that cleanly would be better than manually removing it in this backport.

Reply:
@sueloverso the stats are not a problem, just curious if we should backport those changes first. But this backport still works fine as well! Do you think we should backport WT-8338 first? Or we are greenlit to just merge these changes?

Reply:
If this change will make backporting WT-8338 messier in the future, then backport WT-8338. It should be clean and easy. If that would still be easy then I'm fine just backporting this.

Reply:
Looked at the changes in WT-8338, backporting WT-8338 should still be quite easy after merging this.